plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/projection/DateProjection.java

```diff
@@ -0,0 +1,114 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.aws.athena.projection;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.type.DateType;
+import io.trino.spi.type.TimestampType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.time.Instant;
+import java.time.temporal.ChronoField;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.plugin.hive.aws.athena.DateProjectionFactory.UTC_TIME_ZONE_ID;
+import static io.trino.spi.predicate.Domain.singleValue;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class DateProjection
+        extends Projection
+{
+    private final DateFormat dateFormat;
+    private final Supplier<Instant> leftBound;
+    private final Supplier<Instant> rightBound;
+    private final int interval;
+    private final ChronoUnit intervalUnit;
+
+    public DateProjection(String columnName, DateFormat dateFormat, Supplier<Instant> leftBound, Supplier<Instant> rightBound, int interval, ChronoUnit intervalUnit)
+    {
+        super(columnName);
+        this.dateFormat = requireNonNull(dateFormat, "dateFormatPattern is null");
+        this.leftBound = requireNonNull(leftBound, "leftBound is null");
+        this.rightBound = requireNonNull(rightBound, "rightBound is null");
+        this.interval = interval;
+        this.intervalUnit = requireNonNull(intervalUnit, "intervalUnit is null");
+    }
+
+    @Override
+    public List<String> getProjectedValues(Optional<Domain> partitionValueFilter)
+    {
+        ImmutableList.Builder<String> builder = ImmutableList.builder();
+
+        Instant leftBound = adjustBoundToDateFormat(this.leftBound.get());
+        Instant rightBound = adjustBoundToDateFormat(this.rightBound.get());
+
+        Instant current = leftBound;
+        while (current.isBefore(rightBound) || current.equals(rightBound)) {
+            String formattedValue = formatValue(current);
+            if (isValueInDomain(partitionValueFilter, current, formattedValue)) {
+                builder.add(formattedValue);
+            }
+            current = current.atZone(UTC_TIME_ZONE_ID)
+                    .plus(interval, intervalUnit)
+                    .toInstant();
+        }
```

Comment:
Projection has to assume some time zone. There is no other source of information for zone that should be used by projection, or in which zone given query predicates are considered.
So we always operate on GMT to be sure calculations are correct.

Reply:
After offline discussion with @losipiuk we agreed that handling Date projection is very hard to assure symmetrical  behaviour with AWS Athena. This would required implementing compatibility tests which are a lot of work + additional required resources in AWS account to support them.
So we ended up agreed that for begging we will support only subset of Interval Units that are explicit how they should be handled (DAYS, HOURS, MINUTES, SECONDS).
As when it comes to weeks, months and years it might become tricky and produce different partitions names which will lead to unpredictable results.

Reply:
So we can go with `current = current.plus(interval, intervalUnit)` now, right?